crates/bitcoin-da/src/service.rs

```diff
@@ -168,88 +176,91 @@ impl BitcoinService {
         })
     }
 
-    pub fn spawn_da_queue(
+    pub fn run(
         self: Arc<Self>,
         mut rx: UnboundedReceiver<Option<SenderWithNotifier<TxidWrapper>>>,
     ) {
-        // This is a queue of inscribe requests
-        tokio::task::spawn_blocking(|| {
-            tokio::runtime::Handle::current().block_on(async move {
-                let mut prev_utxo = match self.get_prev_utxo().await {
-                    Ok(Some(prev_utxo)) => Some(prev_utxo),
-                    Ok(None) => {
-                        info!("No pending transactions found");
-                        None
-                    }
-                    Err(e) => {
-                        error!(?e, "Failed to get pending transactions");
-                        None
-                    }
-                };
+        tokio::spawn(async move {
```

Comment:
spawn_blocking might be more suitable for this. 

in prover's case, for instance, mining the wtxid prefix takes a long time. maybe even more than 5 minutes

Reply:
`spawn_blocking` would not work for the async stuff going on inside. We could later update the mining part to be executed inside a blocking context.